dtfs-central-api/src/services/repositories/utilisation-report-status-repo.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Collection,
+  DeleteResult,
+  ObjectId,
+  UpdateResult,
+} from 'mongodb';
+import {
+  ReportDetails,
+  ReportFilter,
+  ReportStatus,
+  UtilisationReport,
+} from '../../types/utilisation-report-status';
+import { TFMUser } from '../../types/users';
+
+const REPORT_STATUS: Record<ReportStatus, ReportStatus> = {
+  RECONCILIATION_COMPLETED: 'RECONCILIATION_COMPLETED',
+  REPORT_NOT_RECEIVED: 'REPORT_NOT_RECEIVED',
+};
+
+const logWarningMessage = (status: ReportStatus) => console.error(`The status '${status}' is not supported by '/v1/utilisation-reports/set-status'`);
+
+const setReportStatusByReportId = (id: string, status: ReportStatus, collection: Collection): Promise<UpdateResult | undefined> => {
+  const filter = { _id: new ObjectId(id) };
+  switch (status) {
+    case REPORT_STATUS.RECONCILIATION_COMPLETED: {
+      return collection.updateOne(filter, {
+        $set: {
+          status: REPORT_STATUS.RECONCILIATION_COMPLETED,
+        },
+      });
+    }
+    case REPORT_STATUS.REPORT_NOT_RECEIVED: {
+      return collection.updateOne(filter, {
+        $set: {
+          status: REPORT_STATUS.REPORT_NOT_RECEIVED,
+        },
+      });
+    }
+    default: {
+      return new Promise<undefined>((resolve) => {
+        logWarningMessage(status);
+        resolve(undefined);
+      });
+    }
+  }
+};
+
+const createOrSetReportAsReceived = (reportDetails: ReportDetails, user: TFMUser, filter: ReportFilter, collection: Collection): Promise<UpdateResult> => {
+  const placeholderUtilisationReport: UtilisationReport = {
+    month: reportDetails.month,
+    year: reportDetails.year,
+    bank: {
+      id: reportDetails.bankId,
+    },
+    azureFileInfo: null,
+    uploadedBy: user,
+    dateUploaded: new Date(),
+  };
```

Comment:
If I want to add the `bank.name` field here, I would need to get the bank from the database using the `bankId` provided. I'm not sure if it is worthwhile to do this for the sake of the placeholder

Reply:
I discussed this with Christian earlier and he did say that we should get the bank name from the database in this case